pkg/autodiscovery/autoconfig.go

```diff
@@ -504,9 +505,34 @@ func (ac *AutoConfig) GetAutodiscoveryErrors() map[string]map[string]providers.E
 // applyChanges applies a configChanges object. This always unschedules first.
 func (ac *AutoConfig) applyChanges(changes configChanges) {
 	if len(changes.unschedule) > 0 {
+		for _, conf := range changes.unschedule {
+			telemetry.ScheduledConfigs.Dec(conf.Provider, configType(conf))
+		}
+
 		ac.scheduler.Unschedule(changes.unschedule)
 	}
+
 	if len(changes.schedule) > 0 {
+		for _, conf := range changes.schedule {
+			telemetry.ScheduledConfigs.Inc(conf.Provider, configType(conf))
+		}
+
 		ac.scheduler.Schedule(changes.schedule)
 	}
 }
+
+func configType(c integration.Config) string {
```

Comment:
Should we move this to the file where `integration.Config` is defined with `integration.Config` as the receiver?

Reply:
It was there and got removed in a recent clean up, probably because it's only really useful here (and even here it looks a bit questionable to me in its current form, but since this is just re-applying an older patch I'm leaving it as is). I have no strong opinions though, so if you still think it's worth the move I'll do it 👍